homeassistant/components/unifi/sensor.py

```diff
@@ -169,6 +178,11 @@ class UnifiSensorEntityDescription(
     """Class describing UniFi sensor entity."""
 
     is_connected_fn: Callable[[UniFiController, str], bool] | None = None
+    # Custom function to determine whether a state change should be recorded
+    value_changed_fn: Callable[
+        [str | int | float | date | Decimal | None, datetime | float | str | None],
```

Comment:
Look at using the same definition as in SensorEntity class `StateType | date | datetime | Decimal`